va/va.h

```diff
@@ -129,6 +129,98 @@ extern "C" {
  * 	- \ref api_fei
  * 	- \ref api_fei_h264
  * 	- \ref api_fei_hevc
+ * 
+ * \section threading Multithreading Guide
+ * All VAAPI functions implemented in libva should be thread-safe, and the 
+ * driver and the backend implementations of VAAPI functions should also be 
+ * thread-safe. However, as VAAPI and libva cannot guarantee the thread-safety
+ * aspect of the driver or the backend implementation, it is ultimately the
+ * responsibility of the driver or the backend implementation to ensure 
+ * thread-saftey of VAAPI functions. 
+ *
+ * It is assumed that none of the VAAPI functions will be called from signal 
+ * handlers.
+ *
+ * Thread-safety in this context means that when VAAPI is being called by multiple
+ * concurrent threads, it will not crash or hang the OS, and VAAPI internal
+ * states will not be corrupted. When multiple threads are operating on the same
+ * VAAPI objects, it is the application's responsibility to synchronize these
+ * operations in order to generate the expected results. This is especially true
+ * when a VAContext object is shared among different threads. Due to the fact that
+ * the internal states of the VAContext object can be modified through API calls  
```

Comment:
I'm confused here. A couple of lines above you say:
> and VAAPI internal states will not be corrupted

but here I read
> the internal states of the VAContext object can be modified

and, presumably, corrupted.  Could you please elaborate on what do you
mean with "`internal states`" in l.145-146 and why VAContext is not one 
of those?

Reply:
Changed "states" to "data structures" which may be a better term. Also removed the block on VAContext as it can just be treated as other VAAPI objects under multi-threaded usage, i.e. app needs to handle the synchronization with operations using VAContext to ensure the correct results.